slsim/lens.py

```diff
@@ -630,6 +664,28 @@ def _point_source_magnitude(self, band, source, lensed=False, time=None):
                 return np.array(magnified_mag_list)
         return source.point_source_magnitude(band)
 
+    def extended_source_magnitude_for_each_images(self, band, lensed=False):
+        """Extended source magnitudes, either unlensed (single value) or lensed
+        (array) with macro-model magnifications. This function provided
```

Comment:
this is an inaccurate description. You need to state that this is when assuming all the light is concentrated at the center of the extended image and magnified as a point source multiple times